src/main/java/com/uvocab/api/service/UserService.java

```diff
@@ -29,4 +34,15 @@ public User findById(long id) {
     }
     throw new NotFoundException("User not found to id: " + id);
   }
+
+  public String login(User user) {
+    var existUser = userRepository.findByLogin(user.getLogin());
+    if (existUser.isPresent()) {
+      var authentication =
+          authenticationManager.authenticate(
```

Comment:
@Danilosimoes we need  only call the authenticationManager.authenticate(), this method do all we need ( **_authenticate the user or throw a exception if user or password is invalid_** ) [here you can see the documentation about this method](https://docs.spring.io/spring-security/site/docs/current/api/org/springframework/security/authentication/AuthenticationManager.html#authenticate(org.springframework.security.core.Authentication))

if the authentication was success only think we need  to do in this moment is return the fixed token

and we need move this method to other class "AuhtService", the same idea of comment controller above